### ISSUE TITLE
Upgrade Rollup to 0.68.1+

### ISSUE THREAD

**Issue by MicahZoltu** [facebook/react#17352]
**Do you want to request a *feature* or report a *bug*?**
Bug-ish.

**What is the current behavior?**
`import 'react'` (where 'react' is the UMD module build) from an ES module will fail due to https://github.com/rollup/rollup/pull/2274.  `this` is not set in the context of ES modules, so if you try to `import 'react'` in that context you will get a runtime error since `this` is undefined and Rollup (before the linked fix) tries to use `this` when it really means `globalThis` or `self`.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than React. Paste the link to your JSFiddle (https://jsfiddle.net/Luktwrdm/) or CodeSandbox (https://codesandbox.io/s/new) example below:**
```
npm install react
```
`index.html`
```html
<!DOCTYPE html>
<head><meta charset='utf-8'></head>
<body>
	<script type='module'>
		import './node_modules/react/umd/react.development.js'
	</script>
</body>
```

**What is the expected behavior?**
The React UMD module can be loaded by an ES module.

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**
React: 16.11.0
Firefox: 70.0.1
Windows 10

**Comment by MicahZoltu** [comment]
This bug is fixed in [Babel 7.7.0](https://github.com/babel/babel/releases/tag/v7.7.0)

**Comment by aweary** [comment]
Is there anything actionable for React here? It seems like this was a Babel problem that is now fixed.

**Comment by MicahZoltu** [comment]
The action item for React here is to upgrade to the new version of Babel so React can be used in ES modules.

Basically, change the version number on these lines to 7.7.0 or higher and then `yarn install` (to generate a new `yarn.lock` file): https://github.com/facebook/react/blob/master/package.json#L7-L34

**Comment by bvaughn** [comment]
Seems like we can remove the "needs info" label. Sounds like the fix is to make sure `@babel/plugin-transform-modules-umd` (at least) gets updated from the current version (7.2) to one with the bugfix (7.7+).

**Comment by bvaughn** [comment]
@MicahZoltu Any interest in picking this up and submitting a PR? :)

**Comment by MicahZoltu** [comment]
@bvaughn PR created, though it has some errors that I don't understand.

**Comment by MicahZoltu** [comment]
Turns out, the issue is actually with an old version of rollup, not babel.  Rollup fixed the bug back in December 2018 with https://github.com/rollup/rollup/pull/2594 (though, that contained a bug that was then rectified in https://github.com/rollup/rollup/pull/2600).

I believe the fix for this is included in all versions of Rollup after 0.68.1, so to resolve this issue React needs to upgrade its Rollup version to 0.68.1.  I tried to update to 0.68.1 but ran into issues, so I think someone more familiar with the React build process will need to take a look at this.

Note: React is now up to version 1+, it may be worth upgrading all the way to 1 (stable release) if someone is going to take the time to bump the rollup version.

**Comment by bvaughn** [comment]
Thanks for the update!

Unfortunately, upgrading certain dependencies (Rollup, Flow) are a little trickier than others. Not sure when we'll get to them, but it is something we discuss from time to time as something we want to get to.

**Comment by alippai** [comment]
This issue is outdated and should be closed. As the current rollup in react is already 1.0+ and the highest available is 2.0+ a new issue would be more appropriate.

**Comment by bvaughn** [comment]
Good call. React is already using Rollup 1.31.1.
